application/dash_application/views/bcl2barcode.py

```diff
@@ -204,65 +202,59 @@ def create_known_index_bar(run):
               data and values for the layout of stacked bar graph of sample indices
               creates bar graph "known_index_bar"
        """
-    data_known = []
-    #   Multiple libraries can use the same index, data must be grouped by both index and libraries
-    #   to prevent duplicate counts
-    for inx, d in run.groupby([COL_INDEX, COL_LIBRARY]):
-        data_known.append(
-            {
-                "x": list(d[COL_LIBRARY].unique()),
-                # One library can be run on multiple lanes. Sum them together.
-                "y": [d[index_col.ReadCount].sum()],
-                "type": "bar",
-                "name": inx[0],
-                "marker": {"line": {"width": 2, "color": "rgb(255,255, 255)"}},
-            }
-        )
+    data = []
+    for i, d in run.groupby(['Sequence', PINERY_COL.SampleName]):
+        data.append({
+            "x": list(d[PINERY_COL.SampleName].unique()),
+            "y": d[bcl2barcode_col.Count],
+            "type": "bar",
+            "name": i[0],
+            "marker": {"line": {"width": 2, "color": "rgb(255,255, 255)"}},
+        })
+    
     return {
-        "data": data_known,
+        "data": data,
         "layout": {
             "barmode": "stack",
             "title": "Sample Indices",
-            "xaxis": {"title": COL_LIBRARY, "automargin": True},
+            "xaxis": {"title": "Library", "automargin": True},
             "yaxis": {"title": "Clusters"},
+            "showlegend": True
         },
     }
 
 
-def create_unknown_index_bar(pruned):
+def create_unknown_index_bar(run):
```

Comment:
Two suggested user improvements:

* Order index by number of unknown reads (those indices with the most unknown should be first)
* No need to show this many unknown indices. I think bcl2fastq showed the top 20.

Reply:
Done